ddtrace/tracer/tracer.go

```diff
@@ -155,6 +162,11 @@ func Start(opts ...StartOption) {
 
 // Stop stops the started tracer. Subsequent calls are valid but become no-op.
 func Stop() {
+	if t, ok := internal.GetGlobalTracer().(*tracer); ok {
```

Comment:
Instead of checking if the global tracer is this type you can just add your stop call to the `tracer.Stop()` function in tracer.go:616